### PR TITLE
Fixes #22373 - Audit manifest import/refresh using Organization

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

**Comment by kgaikwad** [comment]
@ares,
This PR includes audit changes for import/refresh manifest using Organization.

File: test/actions/katello/organization_test.rb [hunk 1]
```diff
@@ -130,6 +130,23 @@ class ManifestRefreshTest < TestBase
...
+      acme_org.stubs(:owner_details).returns({})
+      acme_org.products.stubs(:redhat).returns([rhel7.product])
+      action.stubs(:rand).returns('1234')
+
+      action.stubs(:link!)
+      action.stubs(:lock!)
+      action.stubs(:exclusive_lock!)
+
+      plan_action(action, acme_org)
+      assert_difference 'Audit.count', 1 do
```
Comment (kgaikwad) [inline test/actions/katello/organization_test.rb]: Note - when I have added `acme_org.audits`, it was giving  `<ActiveRecord::Associations::CollectionProxy []>` whereas in `Audit`  getting a new record for Organization.

File: app/lib/actions/katello/organization/manifest_import.rb [hunk 2]
```diff
@@ -26,6 +26,13 @@ def plan(organization, path, force)
         def humanized_name
           _("Import Manifest")
         end
+
+        def finalize
```
Comment (ares) [inline app/lib/actions/katello/organization/manifest_import.rb]: this task does not call `plan_self` in its `plan` method, therefore finalize is not triggered, which explains why I don't see the audit being added or refreshed_at being updated. simply adding plan_self to the end of sequence block should fix it

File: app/lib/actions/katello/organization/manifest_refresh.rb [hunk 3]
```diff
@@ -42,6 +42,13 @@ def rescue_strategy
         def humanized_name
           _("Refresh Manifest")
         end
+
+        def finalize
```
Comment (ares) [inline app/lib/actions/katello/organization/manifest_refresh.rb]: this also does not call plan_self in `plan` method, so finalize is never triggered

File: app/lib/actions/katello/organization/manifest_import.rb (hunk 2 above)
Comment (kgaikwad) [reply app/lib/actions/katello/organization/manifest_import.rb]: Yes, right it doesn't trigger finalize. My mistake that I have tested test cases and I wasn't able to test full flow through UI as on dev setup faced some different issue.
I will fix it asap.

File: db/migrate/20180218191449_add_refreshed_at_to_organization.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class AddRefreshedAtToOrganization < ActiveRecord::Migration[5.1]
+  def up
+    add_column(:taxonomies, :refreshed_at, :datetime)
```
[the remark this answers is not in the document]
Comment (kgaikwad) [reply db/migrate/20180218191449_add_refreshed_at_to_organization.rb]: Yes, I agree.

**Comment by kgaikwad** [comment]
@ares,
Thank you for reviewing a pull-request.
I have corrected code as per review comments.
I was able to test manifest import & refresh actions now I can see audit records.

Added audit changes to `Actions::Katello::Organization::ManifestDelete`. Test case is running fine but when I tried to delete manifest from web-ui, created task for it. That task was running for 3-4 hours and on console, it was continuously showing below request logs. Afterwards, I forcefully stopped my server.

screenshot from 2018-02-24 18-57-35

**Comment by jlsherrill** [comment]
This all works as expected, i did notice its hard to tell the difference between upload/refresh/delete:

audit1

If you click on it you see more info:

audit2

@ares i guess this is the best we can do with the current auditing system? or do you have some other suggestion?

**Comment by ares** [comment]
Sadly that's all we can do for now. It would be great to have ability to set custom label but so far we can only set comments which are displayed on audit detail page.

**Pull request merged**